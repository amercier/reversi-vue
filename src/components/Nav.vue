<template>
  <nav class="nav">
    <ul class="nav-list">
      <li
        class="nav-item"
        v-for="[href, label] in links"
        :key="`${href}-${label}`"
      >
        <router-link class="nav-link" :to="href">{{ label }}</router-link>
      </li>
    </ul>
  </nav>
</template>

<script>
/**
 * Horizontal list of navigation links.
 */
export default {
  name: 'Nav',
  props: {
    /**
     * List of links. Each link consist of an URL (href) and a label.
     *
     * @type {String[][]}
     */
    links: Array,
  },
};
</script>

<style lang="scss">
.nav {
  text-align: center;
  height: 100%;

  &-list {
    list-style: none;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100%;
  }

  &-item {
    display: flex;
    align-items: center;
    list-style: none;
    display: block;
    margin: 0 1em;
  }

  &-link {
    padding: 0.5em 1em;
    white-space: nowrap;

    &.router-link-exact-active {
      font-weight: bold;
      cursor: default;
    }

    &:not(.router-link-exact-active):hover,
    &:not(.router-link-exact-active):focus {
      text-decoration: underline;
    }
  }
}
</style>
