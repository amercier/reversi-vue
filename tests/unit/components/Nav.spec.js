import { shallowMount } from '@vue/test-utils';
import Nav from '@/components/Nav.vue';

describe('Nav.vue', () => {
  const propsData = {
    links: [['/a', 'Link 1'], ['/b', 'Link 2'], ['/c', 'Link 3']],
  };

  it('renders without crashing', () => {
    shallowMount(Nav, { propsData, stubs: ['router-link'] });
  });

  it('renders its components', () => {
    const nav = shallowMount(Nav, { propsData, stubs: ['router-link'] });
    expect(nav.html()).toMatchSnapshot();
  });
});
