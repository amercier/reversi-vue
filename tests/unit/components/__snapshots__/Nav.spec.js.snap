// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav.vue renders its components 1`] = `
<nav
  class="nav"
>
  <ul
    class="nav-list"
  >
    <li
      class="nav-item"
    >
      <router-link-stub
        class="nav-link "
        to="/a"
      >
        Link 1
      </router-link-stub>
       
      <!---->
    </li>
    <li
      class="nav-item"
    >
      <router-link-stub
        class="nav-link "
        to="/b"
      >
        Link 2
      </router-link-stub>
       
      <!---->
    </li>
    <li
      class="nav-item"
    >
      <!---->
       
      <a
        class="nav-link "
        href="#"
      >
        Link 3
      </a>
    </li>
  </ul>
</nav>
`;
